### PR TITLE
#July2019Release1

### PR DESCRIPTION
**Pull request by namrataukirde**
Hi Team,

- Fix rails logging for Splunk. (GCW-2409)
- Start the SMS Reminder Sent At clock AFTER the initial system generated message.
- Free text search of goods. (GCW-2616)

Please review.

Thanks.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -37,12 +37,7 @@ class Application < Rails::Application
 
     config.active_job.queue_adapter = :sidekiq
 
-    config.lograge.enabled = true
-    config.lograge.enabled = Lograge::Formatters::Json.new
-    config.log_formatter = proc do |severity, datetime, progname, msg|
-      "time=\"#{datetime.iso8601(3)}\" level=\"#{severity}\" pid=\"#{Process.pid}\" #{msg.to_s.gsub('"', "'")}\n"
-    end
-
```
Comment (abhinav098) [inline config/application.rb]: is this a side-effect of some mistakenly committed file ? or its intentional ? @namrataukirde

Comment (namrataukirde) [reply config/application.rb]: Let me check @abhinav098. Thanks for mentioning it. :+1:

Comment (namrataukirde) [reply config/application.rb]: Yes, mistakenly deleted I think. We need to re-add it. Thanks :)

Comment (namrataukirde) [reply config/application.rb]: @steveyken, We removed it intentionally or we need it? Please guide us on this.

Comment (patrixr) [reply config/application.rb]:
```
    config.lograge.enabled = true
    config.lograge.enabled = Lograge::Formatters::Json.new
```

this looks wrong to begin with

Comment (steveyken) [reply config/application.rb]: yeah well spotted patrick. it was definitely wrong code to start with but I moved everything to an initializer. see log_formatting.rb

Comment (steveyken) [reply config/application.rb]: and thanks for picking it up @abhinav098

**Pull request merged**